ui/src/app/shared/header/header.component.ts

```diff
@@ -220,6 +221,10 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
     this.navigateWithStatus(this.failedStatuses.slice());
   }
 
+  showOnHoldJobs(): void {
+    this.navigateWithStatus(this.onHoldStatuses.slice());
```

Comment:
It might be nice to represent the statuses with algebraic datatypes instead of lists that we slice around in - i.e. an `Active` trait mixed into classes `Submitted`, `Running`, and`Aborting` but `OnHold` is just a standalone class since there's only one possibility

I don't actually know if this is possible in Typescript, but the "Type" part makes me think yes

Reply:
I think I'd suggest that if one is going to suggest using things which would require actual language constructs that one research the existence of those constructs prior to making the advice :)

Reply:
[Seems like yes](https://www.typescriptlang.org/docs/handbook/mixins.html).

They totally anticipated me coming over from Scala and immediately wanting this in Typescript ("You may be familiar with the idea of mixins or traits for languages like Scala")

Reply:
This was modeled the existing handling of other statuses for code consistency, but I'm open to other implementations.

Reply:
It's not obvious to me what a no-parameter invocation of `slice()` should do, hence my attention. But this is definitely a "thinking out loud" type of discussion rather than something @rsasch needs to change!

Reply:
FYI: It's a fairly common pattern in JS to do a shallow copy of an array.

IIRC there was a bug with naviagateWithStatuses() mutating its parameter. This could probably be cleaned up, e.g. by not mutating the parameter, but I'm foggy on the details.